### PR TITLE
Upgrade to junit:junit:4.13

### PR DESCRIPTION
**Pull request by marcphilipp**
This changes `junit-vintage-engine`, `junit-jupiter-migrationsupport`, and `junit-platform-runner` to compile against 4.13 but adds extra test tasks that run it against 4.12. In the OSGi metadata, we now import the packages from junit:junit with `@4.12` which means `[4.12,5.0)`. That's done because there's no OSGi bundle for 4.13, yet. The published POM and Gradle Module Metadata require 4.13, though, to stop Maven to consume any potential future pre-release (e.g. 4.14-rc-1 or similar).

---

Original description:

> Unfortunately, it doesn't work yet, due to an OSGi error:
> 
> > Task :junit-platform-runner:verifyOSGi FAILED
Resolution failed. Capabilities satisfying the following requirements could not be found:
    [<<INITIAL>>]
      ⇒ osgi.identity: (osgi.identity=junit-platform-runner)
          ⇒ [junit-platform-runner version=1.6.0.SNAPSHOT]
              ⇒ osgi.wiring.package: (&(osgi.wiring.package=org.junit.runner)(version>=4.13.0))
> 
> More details and stacktrace: https://scans.gradle.com/s/er4ytskvf5qqs/failure?openFailures=WzBd&openStackTraces=WzRd#top=0

### CODE REVIEW

**Comment by marcphilipp** [comment]
I believe that's caused by us using `org.apache.servicemix.bundles:org.apache.servicemix.bundles.junit:4.12_1`.

**Comment by marcphilipp** [comment]
I created [SMX4-2000](https://issues.apache.org/jira/browse/SMX4-2000) for a 4.13 bundle. In the meantime, I've relaxed the OSGi package import to allow version [4.12,5.0).

File: buildSrc/src/main/kotlin/osgi-conventions.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+				!org.apiguardian.api,\
+				org.junit.platform.commons.logging;status=INTERNAL,\
+				kotlin.*;resolution:="optional",\
+				*
+
+			# This tells bnd not to complain if a module doesn't actually import
+			# the kotlin packages, but enough modules do to make it a default.
+			-fixupmessages.kotlin.import: "Unused Import-Package instructions: \\[kotlin.*\\]";is:=ignore
+
+			# This tells bnd to ignore classes it files in `META-INF/versions/`
```
Comment (sbrannen) [inline buildSrc/src/main/kotlin/osgi-conventions.gradle.kts]:
```suggestion
			# This tells bnd to ignore classes it finds in `META-INF/versions/`
```

File: buildSrc/src/main/kotlin/osgi-conventions.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+	doLast {
+		// Do the actual work putting OSGi stuff in the jar.
+		btc.buildBundle()
+	}
+}
+
+val osgiPropertiesFile = file("$buildDir/verifyOSGiProperties.bndrun")
+
+// Bnd's Resolve task uses a properties file for it's configuration. This
```
Comment (sbrannen) [inline buildSrc/src/main/kotlin/osgi-conventions.gradle.kts]:
```suggestion
// Bnd's Resolve task uses a properties file for its configuration. This
```

File: buildSrc/src/main/kotlin/osgi-conventions.gradle.kts [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+	property("-runrequires", "osgi.identity;filter:='(osgi.identity=${project.name})'")
+	property("-runsystempackages", "jdk.internal.misc,sun.misc")
+}
+
+val osgiVerification by configurations.creating {
+	extendsFrom(configurations.runtimeClasspath.get())
+}
+
+// Bnd's Resolve task is what verifies that a jar can be used in OSGi and
+// that it's metadata is valid. If the metadata is invalid this task will
```
Comment (sbrannen) [inline buildSrc/src/main/kotlin/osgi-conventions.gradle.kts]:
```suggestion
// that its metadata is valid. If the metadata is invalid this task will
```

File: buildSrc/src/main/kotlin/osgi-conventions.gradle.kts [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+		bundles(osgiVerification)
+	}
+}
+
+tasks.check {
+	dependsOn(verifyOSGi)
+}
+
+// The ${project.description}, for some odd reason, is only available
+// afterEvaluate.
```
Comment (sbrannen) [inline buildSrc/src/main/kotlin/osgi-conventions.gradle.kts]:
```suggestion
// after evaluation.
```

**Comment by marcphilipp** [comment]
> How does it play with Hamcrest 2.2?

Same as before or am I missing sth.?

> Do we have to wait for issues.apache.org/jira/browse/SMX4-2000 being merged?

Nope, since we're importing the `org.junit` packages with `@4.12`.

**Pull request merged**